src/Units/AWBWUnits.java

```diff
@@ -0,0 +1,778 @@
+  }
+
+  @Override
+  public String getIconicUnitName()
+  {
+    return "Infantry";
+  }
+
+  @Override
+  public GameReadyModels getGameReadyModels()
+  {
+    GameReadyModels awbwModels = new GameReadyModels();
+
+    ArrayList<UnitModel> factoryModels = new ArrayList<UnitModel>();
+    ArrayList<UnitModel> seaportModels = new ArrayList<UnitModel>();
+    ArrayList<UnitModel> airportModels = new ArrayList<UnitModel>();
+
+    // Define everything we can build from a Factory.
+    factoryModels.add(new InfantryModel());
+    factoryModels.add(new MechModel());
+    factoryModels.add(new APCModel());
+    factoryModels.add(new ArtilleryModel());
+    factoryModels.add(new ReconModel());
+    factoryModels.add(new TankModel());
+    factoryModels.add(new MDTankModel());
+    factoryModels.add(new NeotankModel());
+    factoryModels.add(new MegatankModel());
+    factoryModels.add(new RocketsModel());
+    factoryModels.add(new AntiAirModel());
+    factoryModels.add(new MobileSAMModel());
+    factoryModels.add(new PiperunnerModel());
+
+    // Record those units we can get from a Seaport.
+    seaportModels.add(new LanderModel());
+    seaportModels.add(new CruiserModel());
+    UnitModel sub = new SubModel();
+    seaportModels.add(sub);
+    seaportModels.add(new BattleshipModel());
+    seaportModels.add(new CarrierModel());
+    seaportModels.add(new BBoatModel());
+
+    // Inscribe those war machines obtainable from an Airport.
+    airportModels.add(new TCopterModel());
+    airportModels.add(new BCopterModel());
+    airportModels.add(new FighterModel());
+    airportModels.add(new BomberModel());
+    UnitModel stealth = new StealthModel();
+    airportModels.add(stealth);
+    airportModels.add(new BBombModel());
+
+    // Dump these lists into a hashmap for easy reference later.
+    awbwModels.shoppingList.put(TerrainType.FACTORY, factoryModels);
+    awbwModels.shoppingList.put(TerrainType.SEAPORT, seaportModels);
+    awbwModels.shoppingList.put(TerrainType.AIRPORT, airportModels);
+
+    // Compile one master list of everything we can build.
+    for (UnitModel um : factoryModels)
+      awbwModels.unitModels.add(um);
+    for (UnitModel um : seaportModels)
+      awbwModels.unitModels.add(um);
+    for (UnitModel um : airportModels)
+      awbwModels.unitModels.add(um);
+
+    // Handle transforming units separately, since we don't want two buy-entries
+    UnitModel subsub = new SubSubModel();
+    sub.possibleActions.add(new UnitActionType.Transform(subsub, "DIVE"));
+    subsub.possibleActions.add(new UnitActionType.Transform(sub, "RISE"));
+    awbwModels.unitModels.add(subsub);
+    UnitModel sneaky = new StealthHideModel();
+    stealth.possibleActions.add(new UnitActionType.Transform(sneaky, "HIDE"));
+    sneaky.possibleActions.add(new UnitActionType.Transform(stealth, "APPEAR"));
+    awbwModels.unitModels.add(sneaky);
+
+    return awbwModels;
+  }
+
+  public enum AWBWUnitEnum
```

Comment:
Wasn't one of your goals to ditch `UnitEnum`? Now commanders don't have a way to find a specific unit type, and `UnitEnum` is duplicated per unit set. Which is maybe unavoidable if we want multiple unit sets, but hey.
It looks like it's only used for indexing into the damage chart though - if only we had a way to avoid needing that...

Reply:
`UnitEnum` is now an implementation detail of any given unit set. You don't need the enum since you don't need a damage chart - only a way to generate a base damage number (which you can calculate by accessing custom variables for your unit set).

I think the damage chart with enum indexing is about as good a system as you can get, for most purposes.
It makes targeted unit damage balance changes trivial, and doesn't require complex logic.